src/osd/OSDMap.cc

```diff
@@ -5197,8 +5198,10 @@ void OSDMap::get_random_up_osds_by_subtree(int n,     // whoami
   if (subtree_type < 1)
     return;
   vector<int> subtrees;
+  std::random_device seed;
+  std::mt19937 rng(seed());  // Start the randomgenerator
```

Comment:
could you consider using `std::default_random_engine` to be explicit that an acceptable RNG is good enough here.